sed/loader/flash/utils.py

```diff
@@ -68,35 +71,39 @@ def get_channels(
             channels.extend(
                 key
                 for key in available_channels
-                if channel_dict[key]["format"] == format_ and key != DLD_AUX_ALIAS
+                if channel_dict[key]["format"] == format_ and key != dld_aux_alias
             )
             # Include 'dldAuxChannels' if the format is 'per_train' and extend_aux is True.
             # Otherwise, include 'dldAux'.
-            if format_ == FORMATS[2] and DLD_AUX_ALIAS in available_channels:
+            if format_ == FORMATS[2] and dld_aux_alias in available_channels:
                 if extend_aux:
                     channels.extend(
-                        channel_dict[DLD_AUX_ALIAS][DLDAUX_CHANNELS].keys(),
+                        channel_dict[dld_aux_alias][aux_subchannels_alias].keys(),
                     )
                 else:
-                    channels.extend([DLD_AUX_ALIAS])
+                    channels.extend([dld_aux_alias])
 
     return channels
 
 
-def get_dtypes(channels_dict: dict, formats: str | list[str]) -> dict:
+def get_dtypes(channels_dict: dict, df_cols: list) -> dict:
     """Returns a dictionary of channels and their corresponding data types.
     Currently Auxiliary channels are not included in the dtype dictionary.
 
     Args:
-        channels_dict (dict): The dictionary containing the channels.
-        formats (str | list[str]): The desired format(s).
+        channels_dict (dict): The config dictionary containing the channels.
+        df_cols (list): A list of channels in the DataFrame.
 
     Returns:
         dict: A dictionary of channels and their corresponding data types.
     """
-    channels = get_channels(channel_dict=channels_dict, formats=formats)
-    return {
-        channel: channels_dict[channel].get("dtype")
-        for channel in channels
-        if channels_dict[channel].get("dtype") is not None
-    }
+    dtypes = {}
+    for channel in df_cols:
+        try:
+            dtypes[channel] = channels_dict[channel].get("dtype")
+        except KeyError:
+            try:
+                dtypes[channel] = channels_dict["dldAux"][channel].get("dtype")
+            except KeyError:
+                dtypes[channel] = None
```

Comment:
Put as default float32 here?

Reply:
I think it's a bit risky if the data is coerced to 32 without users knowledge. I am not sure if a warning is provided by pandas if the data is float64.